### PR TITLE
chore: more arrangement classes

### PR DESCRIPTION
**Pull request by Garzas**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

More arrangement classes needed for federation tests

### CODE REVIEW

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/util/arrangement/repository/ConnectionRepositoryArrangement.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import com.wire.kalium.logic.data.connection.ConnectionRepository
+import com.wire.kalium.logic.data.conversation.ConversationDetails
+import com.wire.kalium.logic.functional.Either
+import io.mockative.Mock
+import io.mockative.given
+import io.mockative.mock
+import kotlinx.coroutines.flow.Flow
+
+internal interface ConnectionRepositoryArrangement {
+    @Mock
```
Comment (vitorhugods) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/util/arrangement/repository/ConnectionRepositoryArrangement.kt:30]:
```suggestion
```

Unnecessary annotation

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/util/arrangement/repository/ConversationRepositoryArrangement.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.util.arrangement.repository
+
+import com.wire.kalium.logic.data.conversation.ConversationRepository
+import io.mockative.Mock
+import io.mockative.mock
+
+internal interface ConversationRepositoryArrangement {
+    @Mock
```
Comment (vitorhugods) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/util/arrangement/repository/ConversationRepositoryArrangement.kt:25]:
```suggestion
```

Unnecessary annotation

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/util/arrangement/repository/UserRepositoryArrangement.kt [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.util.arrangement.repository
+
+import com.wire.kalium.logic.data.user.UserRepository
+import io.mockative.Mock
+import io.mockative.mock
+
+internal interface UserRepositoryArrangement {
+    @Mock
```
Comment (vitorhugods) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/util/arrangement/repository/UserRepositoryArrangement.kt:25]:
```suggestion
```

Unnecessary annotation

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/util/arrangement/usecase/PersistMessageUseCaseArrangement.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.message.PersistMessageUseCase
+import com.wire.kalium.logic.functional.Either
+import io.mockative.Mock
+import io.mockative.any
+import io.mockative.given
+import io.mockative.mock
+
+internal interface PersistMessageUseCaseArrangement {
+    @Mock
```
Comment (vitorhugods) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/util/arrangement/usecase/PersistMessageUseCaseArrangement.kt:29]:
```suggestion
```

Unnecessary annotation

**Review by vitorhugods — APPROVED** [review]
Just small adjustments.

`@Mock` is an unnecessary annotation in abstract definitions.

**Review by alexandreferris — APPROVED** [review]

**Review by yamilmedina — APPROVED** [review]

**Pull request merged**